ion/decimal.go

```diff
@@ -231,21 +231,29 @@ func (d *Decimal) upscale(scale int32) *Decimal {
 	}
 }
 
-// Trunc attempts to truncate this decimal to an int64, dropping any fractional bits.
-func (d *Decimal) Trunc() (int64, error) {
+func (d *Decimal) checkToUpscale() (*Decimal, error) {
 	if d.scale < 0 {
 		// Don't even bother trying this with numbers that *definitely* too big to represent
 		// as an int64, because upscale(0) will consume a bunch of memory.
 		if d.scale < -20 {
-			return 0, &strconv.NumError{
+			return d, &strconv.NumError{
 				Func: "ParseInt",
 				Num:  d.String(),
 				Err:  strconv.ErrRange,
 			}
 		}
 		d = d.upscale(0)
 	}
+	return d, nil
+}
 
+// Trunc attempts to truncate this decimal to an int64, dropping any fractional bits.
+func (d *Decimal) Trunc() (int64, error) {
+	ud, err := d.checkToUpscale()
+	if err != nil {
+		return 0, err
+	}
+	d = ud
 	str := d.n.String()
 
 	want := len(str) - int(d.scale)
```

Comment:
I don't think I understand what `want` is meant to convey here. The variable contains the index of the first rune in the `Decimal`'s `string` representation that we don't want to keep. Maybe `truncateTo`?

Reply:
Updated to `truncatetTo`